Procurement/ViewModel/CurrencyStashViewModel.cs

```diff
@@ -70,35 +70,60 @@ public CurrencyStashViewModel(List<Item> stash)
 
         private ItemDisplayViewModel GetItemAtPosition(int x, int y)
         {
-            var item = _stash.FirstOrDefault(i => i.X == x && i.Y == y);
+            ItemDisplayViewModel rtnViewModel = null;
 
-            return new ItemDisplayViewModel(item);
+            foreach (var item in _stash)
+            {
+                if (item.Key.X == x & item.Key.Y == y)
+                {
+                    rtnViewModel = new ItemDisplayViewModel(item.Key);
+
+                    _stash[item.Key] = rtnViewModel;
```

Comment:
This "getter" function is actually sometimes modifying the class members, which is unexpected.  Also, since it is just setting the value to `new ItemDisplayViewModel(item.Key)`, it is most likely setting the value to something equal to what it was (since that was how it was initialized.  Why not just `return item.Value` here, and a newly-constructed view model if no item is at that position?

``` C#
        private ItemDisplayViewModel GetItemAtPosition(int x, int y)
        {
            foreach (var item in _stash)
            {
                if (item.Key.X == x & item.Key.Y == y)
                {
                    return item.Value;
                }
            }

            return new ItemDisplayViewModel(null);
        }
```

Reply:
You are right - once we've "Found" the POE.Model.Item (Which is the key) we need to give that item it's corresponding ItemDisplayViewModel. This is the collection that we search against when we filter. I've added a clarifying statement to allude to this.

I'm sure there is a cleaner way of doing this but for now I have extracted the common functionality into CommonTabViewModel